src/apphosting/githubConnections.ts

```diff
@@ -117,36 +130,58 @@ export async function linkGitHubRepository(
     installationId = await promptGitHubInstallation(projectId, location, oauthConn);
   }
 
-  let connectionMatchingInstallation = await getConnectionForInstallation(
+  const connectionMatchingInstallation = await getConnectionForInstallation(
     projectId,
     location,
     installationId,
   );
+  if (connectionMatchingInstallation) {
+    const { id: matchingConnectionId } = parseConnectionName(connectionMatchingInstallation.name)!;
 
-  if (!connectionMatchingInstallation) {
-    connectionMatchingInstallation = await createFullyInstalledConnection(
-      projectId,
-      location,
-      generateConnectionId(),
-      oauthConn,
-      installationId,
-    );
+    if (!createConnectionId || matchingConnectionId === createConnectionId) {
```

Comment:
is the `matchingConnectionId === createConnectionId` comparison necessary? We are already checking if a connection with `createConnectionId` already exists on line 102, and returning if so.